dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -625,6 +638,152 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
+    @POST
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    public final Response generateBundle(@Context final HttpServletRequest request,
+                                       @Context final HttpServletResponse response,
+                                       final GenerateBundleForm form) {
+
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource)
+                        .requiredBackendUser(true)
+                        .requestAndResponse(request, response)
+                        .rejectWhenNoUser(true)
+                        .requiredPortlet("publishing-queue")
+                        .init();
+        final User user = initData.getUser();
+        try {
+            final Bundle bundle = APILocator.getBundleAPI().getBundleById(form.bundleId);
+            if (!UtilMethods.isSet(bundle)) {
+                throw new DoesNotExistException("Bundle with ID: " + form.bundleId + " not found");
+            }
+            
+            //set Filter to the bundle
+            final FilterDescriptor filter = APILocator.getPublisherAPI().getFilterDescriptorByKey(form.filterKey);
+            bundle.setFilterKey(filter.getKey());
+            bundle.setOwner(user.getUserId());
+            //set ForcePush value of the filter to the bundle
+            bundle.setForcePush(
+                    (boolean) filter.getFilters().getOrDefault(FilterDescriptor.FORCE_PUSH_KEY,false));
+            bundle.setOperation(form.operation.ordinal());
+            //Update Bundle
+            APILocator.getBundleAPI().updateBundle(bundle);
+
+            //Generate the bundle file for this given operation
+
+            final BundleGenerator generator = new BundleGenerator(bundle, user);
+            
+            DotConcurrentFactory.getInstance().getSubmitter().submit(generator);
+            final String bundleName = bundle.getName().replaceAll("[^\\w.-]", "_");
+            response.setContentType( "application/x-tgz" );
+            response.setHeader( "Content-Disposition", "attachment; filename=" + bundleName  +"-"+ bundle.getId() + ".tar.gz" );
+            final byte whiteByte = 0x20;
+            
+            final long nowTimeMillis = System.currentTimeMillis();
+            
+            // 10 secs
+            final int sleep = 1000;
+
+            // 2 hours
+            int loop = sleep * 60 * 120;
+            try(final OutputStream outStream = response.getOutputStream()){
+                while(generator.bundleFile==null) {
+                    Logger.infoEvery(this.getClass(), "writing bundleId:" + bundle.getId() + " time elapsed:" + Duration.ofMillis(System.currentTimeMillis()-nowTimeMillis).toString().substring(2).replaceAll("(\\d[HMS])(?!$)", "$1 ").toLowerCase(), 10000);
+                    outStream.write(whiteByte);
+                    outStream.flush();
+                    Try.run(()->Thread.sleep(sleep));
+                    if(--loop<=0)break;
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)